boringtun/src/device/peer.rs

```diff
@@ -204,3 +197,31 @@ impl Peer {
         self.index
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::SeedableRng;
+    use x25519_dalek::{PublicKey, StaticSecret};
+
+    // Introduced this test to prevent LLT-5351 recurring in the future:
+    #[test]
+    fn test_connect_endpoint() {
```

Comment:
How does this ensure prevention. If it would be a STREAM socket it would still be opened

Reply:
connect_endpoint() returns an error if the type is Stream, since socket2::socket::new() returns error upon misconfiguration.